File: backend/src/main/java/org/example/backend/domain/postcomment/dto/PostCommentResponseDto.java
package org.example.backend.domain.postcomment.dto;

import org.example.backend.domain.postcomment.entity.PostComment;

public record PostCommentResponseDto(
    Long id,
    Long postId,
    String content,
    String nickname
) {
    public PostCommentResponseDto(PostComment postComment) {
        this(
            postComment.getId(),
            postComment.getPost().getId(),
            postComment.getContent(),
            postComment.getAuthor().getNickname()
        );
    }
}

File: backend/src/main/java/org/example/backend/domain/postcomment/dto/PostCommentResponseDto.kt
package org.example.backend.domain.postcomment.dto

import org.example.backend.domain.postcomment.entity.PostComment

@JvmRecord
data class PostCommentResponseDto(
    val id: Long,
    val postId: Long,
    val content: String,
    val nickname: String
) {
    constructor(postComment: PostComment) : this(
        postComment.id,
        postComment.post.id,
        postComment.content,
        postComment.author.nickname
    )
}
